Mathlib/RingTheory/MvPowerSeries/NoZeroDivisors.lean

```diff
@@ -0,0 +1,74 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir
+-/
+
+import Mathlib.RingTheory.MvPowerSeries.Basic
+import Mathlib.Data.Finsupp.WellFounded
+
+/-! # ZeroDivisors in a MvPowerSeries ring
+
+- `mem_nonZeroDivisors_of_constantCoeff` proves that
+a multivariate power series whose constant coefficient is not a zero divisor
+is itself not a zero divisor
+
+## TODO
+
+Prove that if `R` has no zero divisors, then so does `MvPowerSeries σ R`
+(To avoid induction, it would be nice to endow `σ →₀ ℕ
+with a lexicographic order, for some well ordering of `σ`…)
+
+-/
+
+noncomputable section
+
+open Finset (antidiagonal mem_antidiagonal)
+
+namespace MvPowerSeries
+
+open Finsupp
+
+variable {σ R : Type*}
+
+section Semiring
+
+variable [Semiring R]
+
+/-- A multivariate power series is not a zero divisor
```

Comment:
I could but this looks awkward.
I feel it rather belongs to a PR about HahnSeries (where a lot of things are missing, by the way).

Reply:
I see your point; but it would be a pity if someone had to implement this result for Hahn series and they re-do it from scratch rather than relying on your contribution. A single lemma in the Hahn series file would be in the scope of this PR, I believe, but if you prefer not I would suggest that you add two TODO's both in the mvPS and in the Hahn series file, saying that this could be done but we're waiting for more development of Hahn series.